### PR TITLE
demo app: update thing defs for clib

### PR DESCRIPTION
**Pull request by element-82**
Add a set of thing, app, role definitions for the clib implementation.
The python version has diverged.  Also, update the action.c demo app
to show an alarm and event.

Signed-off-by: Paul Barrette <paulbarrette@gmail.com>

### CODE REVIEW

File: share/admin-tools/thing_defs/hdc_device_manager_def.cfg [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        "name": "Device Manager",
+        "version": 20,
+        "autoDefProps": false,
+        "autoDefAttrs": false,
+        "attributes": {
+        "architecture": {
+            "name": "System Architecture",
+            "default": "Unknown"
+        },
+        "hdc_version": {
```
Comment (lt-holman) [inline share/admin-tools/thing_defs/hdc_device_manager_def.cfg]: api_version in C.

File: share/admin-tools/thing_defs/hdc_device_manager_def.cfg [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            "name": "HDC API Version",
```
Comment (lt-holman) [inline share/admin-tools/thing_defs/hdc_device_manager_def.cfg]: no HDC in name

File: share/admin-tools/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+Usage
+-----
+The default apps and thing defs are maintain in the app_defs/ and
+thing_defs/ directories.  The script will detect if the thing def or
+app is new or existing, and then create or update the definition.
+```sh
+# put your credentials in the env
+HDCADDRESS=core-api.hdcstg.net
```
[the remark this answers is not in the document]
Comment (element-82) [reply share/admin-tools/README.md]: there are 155 occurrences of hdc in our code tree.  I don't want to change the variable names that people are using now in python.

File: share/admin-tools/thing_defs/hdc_device_manager_def.cfg [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        "remote-access": {
+            "name": "Remote Login (For External Trigger)",
+            "notificationVariables": {
+            "host": {
+                "name": "Host Address",
+                "type": "string",
+                "uiType": "text"
+            },
+            "protocol": {
+                "name": "Protocol Port",
```
[the remark this answers is not in the document]
Comment (element-82) [reply share/admin-tools/thing_defs/hdc_device_manager_def.cfg]: Ack

File: share/admin-tools/thing_defs/hdc_device_manager_def.cfg (hunk 1 above)
Comment (element-82) [reply share/admin-tools/thing_defs/hdc_device_manager_def.cfg]: Ack

**Pull request merged**